src/Raven.Client/Http/RequestExecuter.cs

```diff
@@ -296,41 +317,91 @@ private void UpdateTopologyCallback(object _)
         private async Task HandleServerDown<TResult>(ServerNode node, JsonOperationContext context, RavenCommand<TResult> command,
             HttpRequestException e)
         {
-
             if (command.FailedNodes == null)
                 command.FailedNodes = new HashSet<ServerNode>();
 
+            node.IsFailed = true;
             command.FailedNodes.Add(node);
```

Comment:
node.IsFailed = true when command.FailedNodes.Count > 0 ? If yes, maybe the property should auto calculate?